OJS.Workers.ExecutionStrategies/NodeJsPreprocessExecuteAndCheckExecutionStrategy.cs

```diff
@@ -184,26 +185,30 @@ public override ExecutionResult Execute(ExecutionContext executionContext)
 
             // Process the submission and check each test
             var executor = new RestrictedProcessExecutor(this.BaseTimeUsed, this.BaseMemoryUsed);
-            var checker = Checker.CreateChecker(
-                executionContext.CheckerAssemblyName,
-                executionContext.CheckerTypeName,
-                executionContext.CheckerParameter);
 
-            result.TestResults = this.ProcessTests(executionContext, executor, checker, codeSavePath);
+            result.Results = this.ProcessTests(
+                executionContext,
+                executor,
+                executionContext.Input.GetChecker(),
+                codeSavePath);
 
             // Clean up
             File.Delete(codeSavePath);
```

Comment:
We can think of how to abstract this into `IDisposable`. It is very VERY easy to forget this line :/

Reply:
There is a `finally` block after each execution that deletes the whole working directory, but I like the idea of wrapping everything in an IDisposable in [this issue](https://github.com/SoftUni-Internal/suls-issues/issues/5027). General file deletion is better to be done in a Dispose method. There are cases during execution where some files need to be deleted, but others don't and `File.Delete` should remain there for those files.